projects/kit/components/combo-box/combo-box.directive.ts

```diff
@@ -0,0 +1,129 @@
+import type {TuiStringMatcher} from '@taiga-ui/cdk/types';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
+import {tuiIsString} from '@taiga-ui/cdk/utils/miscellaneous';
+import type {TuiDataListAccessor} from '@taiga-ui/core/components/data-list';
+import {tuiAsOptionContent} from '@taiga-ui/core/components/data-list';
+import type {TuiTextfieldAccessor} from '@taiga-ui/core/components/textfield';
+import {
+    tuiAsTextfieldAccessor,
+    tuiInjectAuxiliary,
+    TuiTextfieldComponent,
+    TuiWithTextfield,
+} from '@taiga-ui/core/components/textfield';
+import {tuiDropdownEnabled, tuiDropdownOpen} from '@taiga-ui/core/directives/dropdown';
+import type {TuiItemsHandlers} from '@taiga-ui/core/directives/items-handlers';
+import {TUI_ITEMS_HANDLERS} from '@taiga-ui/core/directives/items-handlers';
+import {tuiAsAuxiliary} from '@taiga-ui/core/tokens';
+import {TuiSelectOption} from '@taiga-ui/kit/components/select';
+
+@Directive({
+    standalone: true,
+    selector: 'input[tuiComboBox]',
+    providers: [
+        tuiAsOptionContent(TuiSelectOption),
+        tuiAsTextfieldAccessor(TuiComboBox),
+        tuiAsControl(TuiComboBox),
+        tuiAsAuxiliary(TuiComboBox),
+    ],
+    hostDirectives: [TuiWithTextfield],
+    host: {
+        '[disabled]': 'disabled()',
+        '(blur)': 'onTouched()',
+        '(click)': 'toggle()',
+        '(input)': 'onInput($event.target.value)',
+        '(keydown.enter)': 'keydownEnter($event)',
+    },
+})
+export class TuiComboBox<T>
+    extends TuiControl<T | string | null>
+    implements TuiTextfieldAccessor<T>
+{
+    private readonly el = tuiInjectElement<HTMLInputElement>();
+    private readonly textfield: TuiTextfieldComponent<T> = inject(TuiTextfieldComponent);
+
+    private readonly open = tuiDropdownOpen();
+    private readonly itemsHandlers: TuiItemsHandlers<T> = inject(TUI_ITEMS_HANDLERS);
+    private readonly datalist = tuiInjectAuxiliary<TuiDataListAccessor<T>>(
+        (x) => 'getOptions' in x,
+    );
+
+    protected readonly dropdownEnabled = tuiDropdownEnabled(this.interactive);
+
+    protected readonly valueEffect = effect(() => {
+        const value = this.value() ?? '';
+        const stringified = tuiIsString(value)
+            ? value
+            : this.itemsHandlers.stringify()(value);
+
+        this.textfield.value.update((x) => stringified || x);
+    }, TUI_ALLOW_SIGNAL_WRITES);
+
+    protected readonly blurEffect = effect(() => {
+        const value = untracked(() => this.value());
+
+        if (!this.textfield.focused() && this.strict && !value) {
+            this.textfield.value.set('');
+        }
+    }, TUI_ALLOW_SIGNAL_WRITES);
+
+    @Input()
+    public strict = true;
+
+    // TODO before merge: `strictMatcher` is used not only for `[strict]=true`.
+    //  Should we rename it to `searchMatcher` ?
+    @Input()
+    public strictMatcher: TuiStringMatcher<T> | null = TUI_STRICT_MATCHER;
```

Comment:
`strictMatcher` is used not only for `[strict]=true`.
Should we rename it to `searchMatcher` ?

Reply:
It was renamed to just `matcher` https://github.com/taiga-family/taiga-ui/pull/10980/commits/305b7914b523cd383e063fccf8c11c58e58a3022